src/components/InstanceProvider.tsx

```diff
@@ -1,14 +1,27 @@
 import React, { useState, FC, useEffect } from 'react';
 import { InstanceContext } from 'components/InstanceContext';
 import { GlobalSettings, GrafanaInstances } from 'types';
-import { config, getDataSourceSrv } from '@grafana/runtime';
+import { config, getDataSourceSrv, getBackendSrv } from '@grafana/runtime';
 import { SMDataSource } from 'datasource/DataSource';
 import { Spinner } from '@grafana/ui';
 import { AppPluginMeta } from '@grafana/data';
 import { UnprovisionedSetup } from 'components/UnprovisionedSetup';
 
-function getRulerDatasource() {
-  return Object.values(config.datasources).find(ds => ds.type === 'grafana-ruler-datasource');
+async function getRulerDatasource(metricDatasourceId?: number) {
+  if (!metricDatasourceId) {
+    return undefined;
+  }
+  const basicAuthUserId = await getBackendSrv()
+    .get(`api/datasources/${metricDatasourceId}`)
+    .then(settings => settings.basicAuthUser);
+  const rulers = Object.values(config.datasources).filter(ds => ds.type === 'grafana-ruler-datasource');
+  const rulerSettings = await Promise.all(
+    rulers.map(ruler => {
+      return getBackendSrv().get(`api/datasources/${ruler.id}`);
+    })
+  );
```

Comment:
You've probably already considered it but this might result in a whole bunch of requests being made, so it might be worthwhile not doing it on the fly every time. In the Alerting UI plugin, we collect all that info when the tab is mounted.
But up to you of course. :)